### ISSUE TITLE
Lots of errors in Debian 10 x86_64 or ARM (Raspberry PI) install

### ISSUE THREAD

**Issue by craigusus** [diladele/websafety#1391]
End of running 03_squid.sh...
```
collect2: error: ld returned 1 exit status
libtool: link: rm -f ".libs/squidS.o"
make[4]: *** [Makefile:6356: squid] Error 1
make[4]: Leaving directory '/home/pi/websafety/core.debian10/build/squid/squid-4.9/src'
make[3]: *** [Makefile:7126: all-recursive] Error 1
make[3]: Leaving directory '/home/pi/websafety/core.debian10/build/squid/squid-4.9/src'
make[2]: *** [Makefile:6136: all] Error 2
make[2]: Leaving directory '/home/pi/websafety/core.debian10/build/squid/squid-4.9/src'
make[1]: *** [Makefile:584: all-recursive] Error 1
make[1]: Leaving directory '/home/pi/websafety/core.debian10/build/squid/squid-4.9'
make: *** [/usr/share/cdbs/1/class/makefile.mk:77: debian/stamp-makefile-build] Error 2
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
~/websafety/core.debian10
```

Running 04_install.sh... (says 05_install.sh in docs, incorrect!)
```
pi@squid:~/websafety/core.debian10 $ sudo ./04_install.sh 
/home/pi/websafety/core.debian10/build/squid /home/pi/websafety/core.debian10
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  squid-langpack
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 167 kB of archives.
After this operation, 4,108 kB of additional disk space will be used.
Get:1 http://mirror.mythic-beasts.com/raspbian/raspbian buster/main armhf squid-langpack all 20190110-1 [167 kB]
Fetched 167 kB in 0s (430 kB/s)        
Selecting previously unselected package squid-langpack.
(Reading database ... 102371 files and directories currently installed.)
Preparing to unpack .../squid-langpack_20190110-1_all.deb ...
Unpacking squid-langpack (20190110-1) ...
Setting up squid-langpack (20190110-1) ...
dpkg: error: cannot access archive 'squid-common_4.9-2_all.deb': No such file or directory
dpkg: error: cannot access archive 'squid_4.9-2_armhf.deb': No such file or directory
dpkg: error: cannot access archive 'squidclient_4.9-2_armhf.deb': No such file or directory
/home/pi/websafety/core.debian10
squid set on hold.
squidclient set on hold.
squid-common set on hold.
squid-langpack set on hold.
rm: cannot remove '/etc/systemd/system/squid.service.d/override.conf': No such file or directory
```

Running 06_websafety.sh...
```
pi@squid:~/websafety/core.debian10 $ sudo ./06_websafety.sh 
--2019-12-24 09:21:19--  http://packages.diladele.com/websafety-core/7.2.0.01D7/armhf/release/debian10/websafety-7.2.0.01D7_armhf.deb
Resolving packages.diladele.com (packages.diladele.com)... 78.47.119.33
Connecting to packages.diladele.com (packages.diladele.com)|78.47.119.33|:80... connected.
HTTP request sent, awaiting response... 404 Not Found
2019-12-24 09:21:20 ERROR 404: Not Found.

dpkg: error: cannot access archive 'websafety-7.2.0.01D7_armhf.deb': No such file or directory
chown: cannot access '/opt/websafety': No such file or directory
```

**Comment by craigusus** [comment]
Also,please can you check the docs [https://docs.diladele.com/administrator_guide_stable/install/debian10/index.html](url)
Step 4 says '05_install.sh script' there is no 05_install.sh script. Its '04_install.sh'.

**Comment by ra-at-diladele-com** [comment]
thanks! I will take a look and get back to you.. might be some unsync in the docs and actual scripts :(

**Comment by craigusus** [comment]
Great thanks. Willing to help run some tests if needed.

**Comment by ra-at-diladele-com** [comment]
If possible please just run all scripts from https://github.com/diladele/websafety/tree/master/core.debian10  then from https://github.com/diladele/websafety/tree/master/ui.deb  on the RPI to see if anything is missing.

**Comment by craigusus** [comment]
I have just started from a clean build and still getting the errors:

When running 03_squid.sh
```
/usr/include/c++/8/bits/atomic_base.h:374: undefined reference to `__atomic_store_8'
collect2: error: ld returned 1 exit status
libtool: link: rm -f ".libs/squidS.o"
make[4]: *** [Makefile:6356: squid] Error 1
make[4]: Leaving directory '/home/pi/websafety/core.debian10/build/squid/squid-4.9/src'
make[3]: *** [Makefile:7126: all-recursive] Error 1
make[3]: Leaving directory '/home/pi/websafety/core.debian10/build/squid/squid-4.9/src'
make[2]: *** [Makefile:6136: all] Error 2
make[2]: Leaving directory '/home/pi/websafety/core.debian10/build/squid/squid-4.9/src'
make[1]: *** [Makefile:584: all-recursive] Error 1
make[1]: Leaving directory '/home/pi/websafety/core.debian10/build/squid/squid-4.9'
make: *** [/usr/share/cdbs/1/class/makefile.mk:77: debian/stamp-makefile-build] Error 2
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
~/websafety/core.debian10
```
When running 04_install.sh
```
pi@raspberrypi:~/websafety/core.debian10 $ sudo ./04_install.sh 
/home/pi/websafety/core.debian10/build/squid /home/pi/websafety/core.debian10
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  squid-langpack
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 167 kB of archives.
After this operation, 4,108 kB of additional disk space will be used.
Get:1 http://raspbian.mirror.uk.sargasso.net/raspbian buster/main armhf squid-langpack all 20190110-1 [167 kB]
Fetched 167 kB in 0s (393 kB/s)        
Selecting previously unselected package squid-langpack.
(Reading database ... 102371 files and directories currently installed.)
Preparing to unpack .../squid-langpack_20190110-1_all.deb ...
Unpacking squid-langpack (20190110-1) ...
Setting up squid-langpack (20190110-1) ...
dpkg: error: cannot access archive 'squid-common_4.9-2_all.deb': No such file or directory
dpkg: error: cannot access archive 'squid_4.9-2_armhf.deb': No such file or directory
dpkg: error: cannot access archive 'squidclient_4.9-2_armhf.deb': No such file or directory
/home/pi/websafety/core.debian10
squid set on hold.
squidclient set on hold.
squid-common set on hold.
squid-langpack set on hold.
rm: cannot remove '/etc/systemd/system/squid.service.d/override.conf': No such file or directory
pi@raspberrypi:~/websafety/core.debian10 $
```

The above .deb files are nowhere to be found on my Pi
```
pi@raspberrypi:~/websafety/core.debian10 $ locate *.deb
/home/pi/websafety/ui.deb
/var/cache/apt/archives/at_3.1.23-1_armhf.deb
/var/cache/apt/archives/autoconf_2.69-11_all.deb
/var/cache/apt/archives/automake_1%3a1.16.1-4_all.deb
/var/cache/apt/archives/autopoint_0.19.8.1-9_all.deb
/var/cache/apt/archives/autotools-dev_20180224.1_all.deb
/var/cache/apt/archives/cdbs_0.4.159_all.deb
/var/cache/apt/archives/comerr-dev_2.1-1.44.5-1+deb10u2_armhf.deb
/var/cache/apt/archives/dctrl-tools_2.24-3_armhf.deb
/var/cache/apt/archives/debhelper_12.1.1_all.deb
/var/cache/apt/archives/debian-keyring_2019.02.25_all.deb
/var/cache/apt/archives/devscripts_2.19.5+deb10u1_armhf.deb
/var/cache/apt/archives/dh-apparmor_2.13.2-10_all.deb
/var/cache/apt/archives/dh-autoreconf_19_all.deb
/var/cache/apt/archives/dh-strip-nondeterminism_1.1.2-1_all.deb
/var/cache/apt/archives/diffstat_1.62-1_armhf.deb
/var/cache/apt/archives/dput_1.0.3_all.deb
/var/cache/apt/archives/dwz_0.12-3_armhf.deb
/var/cache/apt/archives/equivs_2.2.0_all.deb
/var/cache/apt/archives/exim4-base_4.92-8+deb10u3_armhf.deb
/var/cache/apt/archives/exim4-config_4.92-8+deb10u3_all.deb
/var/cache/apt/archives/exim4-daemon-light_4.92-8+deb10u3_armhf.deb
/var/cache/apt/archives/gettext_0.19.8.1-9_armhf.deb
/var/cache/apt/archives/guile-2.2-libs_2.2.4+1-2_armhf.deb
/var/cache/apt/archives/icu-devtools_63.1-6_armhf.deb
/var/cache/apt/archives/intltool-debian_0.35.0+20060710.5_all.deb
/var/cache/apt/archives/krb5-multidev_1.17-3_armhf.deb
/var/cache/apt/archives/libapt-pkg-perl_0.1.34+b1_armhf.deb
/var/cache/apt/archives/libarchive-cpio-perl_0.10-1_all.deb
/var/cache/apt/archives/libarchive-zip-perl_1.64-1_all.deb
/var/cache/apt/archives/libasync-mergepoint-perl_0.04-2_all.deb
/var/cache/apt/archives/libauthen-sasl-perl_2.1600-1_all.deb
/var/cache/apt/archives/libb-hooks-endofscope-perl_0.24-1_all.deb
/var/cache/apt/archives/libb-hooks-op-check-perl_0.22-1+b1_armhf.deb
/var/cache/apt/archives/libcap-dev_1%3a2.25-2_armhf.deb
/var/cache/apt/archives/libcapture-tiny-perl_0.48-1_all.deb
/var/cache/apt/archives/libcgi-fast-perl_1%3a2.13-1_all.deb
/var/cache/apt/archives/libcgi-pm-perl_4.40-1_all.deb
/var/cache/apt/archives/libclass-accessor-perl_0.51-1_all.deb
/var/cache/apt/archives/libclass-inspector-perl_1.32-1_all.deb
/var/cache/apt/archives/libclass-method-modifiers-perl_2.12-1_all.deb
/var/cache/apt/archives/libclass-xsaccessor-perl_1.19-3+b6_armhf.deb
/var/cache/apt/archives/libclone-perl_0.41-1+b1_armhf.deb
/var/cache/apt/archives/libcommon-sense-perl_3.74-2+b18_armhf.deb
/var/cache/apt/archives/libconst-fast-perl_0.014-1_all.deb
/var/cache/apt/archives/libcontextual-return-perl_0.004014-2_all.deb
/var/cache/apt/archives/libconvert-binhex-perl_1.125-1_all.deb
/var/cache/apt/archives/libcppunit-1.14-0_1.14.0-3_armhf.deb
/var/cache/apt/archives/libcppunit-dev_1.14.0-3_armhf.deb
/var/cache/apt/archives/libdata-dump-perl_1.23-1_all.deb
/var/cache/apt/archives/libdata-optlist-perl_0.110-1_all.deb
/var/cache/apt/archives/libdb-dev_5.3.1+nmu1_armhf.deb
/var/cache/apt/archives/libdb5.3-dev_5.3.28+dfsg1-0.5_armhf.deb
/var/cache/apt/archives/libdbi-perl_1.642-1+b1_armhf.deb
/var/cache/apt/archives/libdevel-callchecker-perl_0.008-1_armhf.deb
/var/cache/apt/archives/libdevel-globaldestruction-perl_0.14-1_all.deb
/var/cache/apt/archives/libdigest-bubblebabble-perl_0.02-2_all.deb
/var/cache/apt/archives/libdigest-hmac-perl_1.03+dfsg-2_all.deb
/var/cache/apt/archives/libdistro-info-perl_0.21_all.deb
/var/cache/apt/archives/libdynaloader-functions-perl_0.003-1_all.deb
/var/cache/apt/archives/libecap3-dev_1.0.1-3.2+rpi1_armhf.deb
/var/cache/apt/archives/libecap3_1.0.1-3.2+rpi1_armhf.deb
/var/cache/apt/archives/libemail-valid-perl_1.202-1_all.deb
/var/cache/apt/archives/libencode-locale-perl_1.05-1_all.deb
/var/cache/apt/archives/libexporter-tiny-perl_1.002001-1_all.deb
/var/cache/apt/archives/libfcgi-perl_0.78-2+b2_armhf.deb
/var/cache/apt/archives/libfile-basedir-perl_0.08-1_all.deb
/var/cache/apt/archives/libfile-chdir-perl_0.1008-1_all.deb
/var/cache/apt/archives/libfile-homedir-perl_1.004-1_all.deb
/var/cache/apt/archives/libfile-listing-perl_6.04-1_all.deb
/var/cache/apt/archives/libfile-stripnondeterminism-perl_1.1.2-1_all.deb
/var/cache/apt/archives/libfile-which-perl_1.23-1_all.deb
/var/cache/apt/archives/libfl2_2.6.4-6.2_armhf.deb
/var/cache/apt/archives/libfont-afm-perl_1.20-2_all.deb
/var/cache/apt/archives/libfuture-perl_0.39-1_all.deb
/var/cache/apt/archives/libgc1c2_1%3a7.6.4-0.4_armhf.deb
/var/cache/apt/archives/libgetopt-long-descriptive-perl_0.103-2_all.deb
/var/cache/apt/archives/libgit-wrapper-perl_0.048-1_all.deb
/var/cache/apt/archives/libgitlab-api-v4-perl_0.16-1_all.deb
/var/cache/apt/archives/libgmp-dev_2%3a6.1.2+dfsg-4_armhf.deb
/var/cache/apt/archives/libgmpxx4ldbl_2%3a6.1.2+dfsg-4_armhf.deb
/var/cache/apt/archives/libgnutls-dane0_3.6.7-4_armhf.deb
/var/cache/apt/archives/libgnutls-openssl27_3.6.7-4_armhf.deb
/var/cache/apt/archives/libgnutls28-dev_3.6.7-4_armhf.deb
/var/cache/apt/archives/libgnutlsxx28_3.6.7-4_armhf.deb
/var/cache/apt/archives/libgpgme11_1.12.0-6_armhf.deb
/var/cache/apt/archives/libgsasl7_1.8.0-8+b1_armhf.deb
/var/cache/apt/archives/libgssrpc4_1.17-3_armhf.deb
/var/cache/apt/archives/libhtml-form-perl_6.03-1_all.deb
/var/cache/apt/archives/libhtml-format-perl_2.12-1_all.deb
/var/cache/apt/archives/libhtml-parser-perl_3.72-3+b2_armhf.deb
/var/cache/apt/archives/libhtml-tagset-perl_3.20-3_all.deb
/var/cache/apt/archives/libhtml-tree-perl_5.07-2_all.deb
/var/cache/apt/archives/libhttp-cookies-perl_6.04-1_all.deb
/var/cache/apt/archives/libhttp-daemon-perl_6.01-3_all.deb
/var/cache/apt/archives/libhttp-date-perl_6.02-1_all.deb
/var/cache/apt/archives/libhttp-message-perl_6.18-1_all.deb
/var/cache/apt/archives/libhttp-negotiate-perl_6.01-1_all.deb
/var/cache/apt/archives/libhttp-tiny-multipart-perl_0.08-1_all.deb
/var/cache/apt/archives/libicu-dev_63.1-6_armhf.deb
/var/cache/apt/archives/libidn2-dev_2.0.5-1_armhf.deb
/var/cache/apt/archives/libimport-into-perl_1.002005-1_all.deb
/var/cache/apt/archives/libio-async-perl_0.72-1_all.deb
/var/cache/apt/archives/libio-html-perl_1.001-1_all.deb
/var/cache/apt/archives/libio-prompter-perl_0.004015-1_all.deb
/var/cache/apt/archives/libio-pty-perl_1%3a1.08-1.1+b4_armhf.deb
/var/cache/apt/archives/libio-sessiondata-perl_1.03-1_all.deb
/var/cache/apt/archives/libio-socket-ssl-perl_2.060-3_all.deb
/var/cache/apt/archives/libio-string-perl_1.08-3_all.deb
/var/cache/apt/archives/libio-stringy-perl_2.111-3_all.deb
/var/cache/apt/archives/libipc-run-perl_20180523.0-1_all.deb
/var/cache/apt/archives/libipc-system-simple-perl_1.25-4_all.deb
/var/cache/apt/archives/libjson-perl_4.02000-1_all.deb
/var/cache/apt/archives/libjson-xs-perl_3.040-1+b1_armhf.deb
/var/cache/apt/archives/libkadm5clnt-mit11_1.17-3_armhf.deb
/var/cache/apt/archives/libkadm5srv-mit11_1.17-3_armhf.deb
/var/cache/apt/archives/libkdb5-9_1.17-3_armhf.deb
/var/cache/apt/archives/libkrb5-dev_1.17-3_armhf.deb
/var/cache/apt/archives/libkyotocabinet16v5_1.2.76-4.2+rpi1_armhf.deb
/var/cache/apt/archives/libldap2-dev_2.4.47+dfsg-3+rpi1+deb10u1_armhf.deb
/var/cache/apt/archives/liblist-compare-perl_0.53-1_all.deb
/var/cache/apt/archives/liblist-moreutils-perl_0.416-1+b3_armhf.deb
/var/cache/apt/archives/liblog-any-adapter-screen-perl_0.140-1_all.deb
/var/cache/apt/archives/liblog-any-perl_1.707-1_all.deb
/var/cache/apt/archives/libltdl-dev_2.4.6-9_armhf.deb
/var/cache/apt/archives/liblwp-mediatypes-perl_6.02-1_all.deb
/var/cache/apt/archives/liblwp-protocol-https-perl_6.07-2_all.deb
/var/cache/apt/archives/liblzo2-2_2.10-0.1_armhf.deb
/var/cache/apt/archives/libmail-sendmail-perl_0.80-1_all.deb
/var/cache/apt/archives/libmailtools-perl_2.18-1_all.deb
/var/cache/apt/archives/libmailutils5_1%3a3.5-3_armhf.deb
/var/cache/apt/archives/libmariadb3_1%3a10.3.17-0+deb10u1_armhf.deb
/var/cache/apt/archives/libmime-tools-perl_5.509-1_all.deb
/var/cache/apt/archives/libmodule-implementation-perl_0.09-1_all.deb
/var/cache/apt/archives/libmodule-runtime-perl_0.016-1_all.deb
/var/cache/apt/archives/libmoo-perl_2.003004-2_all.deb
/var/cache/apt/archives/libnamespace-clean-perl_0.27-1_all.deb
/var/cache/apt/archives/libnet-dns-perl_1.19-1_all.deb
/var/cache/apt/archives/libnet-dns-sec-perl_1.11-1_armhf.deb
/var/cache/apt/archives/libnet-domain-tld-perl_1.75-1_all.deb
/var/cache/apt/archives/libnet-http-perl_6.18-1_all.deb
/var/cache/apt/archives/libnet-ip-perl_1.26-2_all.deb
/var/cache/apt/archives/libnet-libidn-perl_0.12.ds-3+b1_armhf.deb
/var/cache/apt/archives/libnet-smtp-ssl-perl_1.04-1_all.deb
/var/cache/apt/archives/libnet-ssleay-perl_1.85-2+b1_armhf.deb
/var/cache/apt/archives/libnetfilter-conntrack-dev_1.0.7-1_armhf.deb
/var/cache/apt/archives/libnfnetlink-dev_1.0.1-3_armhf.deb
/var/cache/apt/archives/libntlm0_1.5-1_armhf.deb
/var/cache/apt/archives/libnumber-compare-perl_0.03-1_all.deb
/var/cache/apt/archives/libnumber-range-perl_0.12-1_all.deb
/var/cache/apt/archives/libossp-uuid-perl_1.6.2-1.5+b5_armhf.deb
/var/cache/apt/archives/libp11-kit-dev_0.23.15-2_armhf.deb
/var/cache/apt/archives/libpackage-stash-perl_0.38-1_all.deb
/var/cache/apt/archives/libpackage-stash-xs-perl_0.29-1_armhf.deb
/var/cache/apt/archives/libpam0g-dev_1.3.1-5+rpt1_armhf.deb
/var/cache/apt/archives/libparams-classify-perl_0.015-1+b1_armhf.deb
/var/cache/apt/archives/libparams-util-perl_1.07-3+b3_armhf.deb
/var/cache/apt/archives/libparams-validate-perl_1.29-1+b1_armhf.deb
/var/cache/apt/archives/libparse-debianchangelog-perl_1.2.0-13_all.deb
/var/cache/apt/archives/libpath-iterator-rule-perl_1.014-1_all.deb
/var/cache/apt/archives/libpath-tiny-perl_0.108-1_all.deb
/var/cache/apt/archives/libperlio-gzip-perl_0.19-1+b4_armhf.deb
/var/cache/apt/archives/libpod-constants-perl_0.19-1_all.deb
/var/cache/apt/archives/libreadonly-perl_2.050-1_all.deb
/var/cache/apt/archives/libref-util-perl_0.204-1_all.deb
/var/cache/apt/archives/libref-util-xs-perl_0.117-1+b1_armhf.deb
/var/cache/apt/archives/libregexp-pattern-license-perl_3.0.31-4_all.deb
/var/cache/apt/archives/librole-tiny-perl_2.000006-1_all.deb
/var/cache/apt/archives/libsasl2-dev_2.1.27+dfsg-1+deb10u1_armhf.deb
/var/cache/apt/archives/libsereal-decoder-perl_4.005+ds-1+b1_armhf.deb
/var/cache/apt/archives/libsereal-encoder-perl_4.005+ds-1+b1_armhf.deb
/var/cache/apt/archives/libsereal-perl_4.005-1_all.deb
/var/cache/apt/archives/libsigsegv2_2.12-2_armhf.deb
/var/cache/apt/archives/libsoap-lite-perl_1.27-1_all.deb
/var/cache/apt/archives/libsort-key-perl_1.33-2+b1_armhf.deb
/var/cache/apt/archives/libsort-versions-perl_1.62-1_all.deb
/var/cache/apt/archives/libssl-dev_1.1.1d-0+deb10u2_armhf.deb
/var/cache/apt/archives/libstrictures-perl_2.000005-1_all.deb
/var/cache/apt/archives/libstring-copyright-perl_0.003006-1_all.deb
/var/cache/apt/archives/libstring-escape-perl_2010.002-2_all.deb
/var/cache/apt/archives/libstring-shellquote-perl_1.04-1_all.deb
/var/cache/apt/archives/libstruct-dumb-perl_0.09-1_all.deb
/var/cache/apt/archives/libsub-exporter-perl_0.987-1_all.deb
/var/cache/apt/archives/libsub-exporter-progressive-perl_0.001013-1_all.deb
/var/cache/apt/archives/libsub-identify-perl_0.14-1+b1_armhf.deb
/var/cache/apt/archives/libsub-install-perl_0.928-1_all.deb
/var/cache/apt/archives/libsub-name-perl_0.21-1+b2_armhf.deb
/var/cache/apt/archives/libsub-quote-perl_2.005001-1_all.deb
/var/cache/apt/archives/libsys-hostname-long-perl_1.5-1_all.deb
/var/cache/apt/archives/libtask-weaken-perl_1.06-1_all.deb
/var/cache/apt/archives/libtasn1-6-dev_4.13-3_armhf.deb
/var/cache/apt/archives/libtasn1-doc_4.13-3_all.deb
/var/cache/apt/archives/libterm-readkey-perl_2.38-1_armhf.deb
/var/cache/apt/archives/libtest-fatal-perl_0.014-1_all.deb
/var/cache/apt/archives/libtest-refcount-perl_0.08-3_all.deb
/var/cache/apt/archives/libtext-glob-perl_0.10-1_all.deb
/var/cache/apt/archives/libtext-levenshtein-perl_0.13-1_all.deb
/var/cache/apt/archives/libtimedate-perl_2.3000-2_all.deb
/var/cache/apt/archives/libtool_2.4.6-9_all.deb
/var/cache/apt/archives/libtry-tiny-perl_0.30-1_all.deb
/var/cache/apt/archives/libtype-tiny-perl_1.004004-1_all.deb
/var/cache/apt/archives/libtype-tiny-xs-perl_0.014-1+b1_armhf.deb
/var/cache/apt/archives/libtypes-serialiser-perl_1.0-1_all.deb
/var/cache/apt/archives/libunbound8_1.9.0-2+deb10u1_armhf.deb
/var/cache/apt/archives/libunicode-utf8-perl_0.62-1_armhf.deb
/var/cache/apt/archives/liburi-perl_1.76-1_all.deb
/var/cache/apt/archives/libvariable-magic-perl_0.62-1+b1_armhf.deb
/var/cache/apt/archives/libwant-perl_0.29-1+b3_armhf.deb
/var/cache/apt/archives/libwww-perl_6.36-2_all.deb
/var/cache/apt/archives/libwww-robotrules-perl_6.02-1_all.deb
/var/cache/apt/archives/libxml-libxml-perl_2.0134+dfsg-1_armhf.deb
/var/cache/apt/archives/libxml-namespacesupport-perl_1.12-1_all.deb
/var/cache/apt/archives/libxml-parser-perl_2.44-4_armhf.deb
/var/cache/apt/archives/libxml-sax-base-perl_1.09-1_all.deb
/var/cache/apt/archives/libxml-sax-expat-perl_0.51-1_all.deb
/var/cache/apt/archives/libxml-sax-perl_1.00+dfsg-1_all.deb
/var/cache/apt/archives/libxml-simple-perl_2.25-1_all.deb
/var/cache/apt/archives/libxml2-dev_2.9.4+dfsg1-7+b3_armhf.deb
/var/cache/apt/archives/libxmlrpc-lite-perl_0.717-2_all.deb
/var/cache/apt/archives/libyaml-libyaml-perl_0.76+repack-1_armhf.deb
/var/cache/apt/archives/licensecheck_3.0.31-3_all.deb
/var/cache/apt/archives/lintian_2.15.0_all.deb
/var/cache/apt/archives/m4_1.4.18-2_armhf.deb
/var/cache/apt/archives/mailutils-common_1%3a3.5-3_all.deb
/var/cache/apt/archives/mailutils_1%3a3.5-3_armhf.deb
/var/cache/apt/archives/mariadb-common_1%3a10.3.17-0+deb10u1_all.deb
/var/cache/apt/archives/mysql-common_5.8+1.0.5_all.deb
/var/cache/apt/archives/nettle-dev_3.4.1-1_armhf.deb
/var/cache/apt/archives/patchutils_0.3.4-2_armhf.deb
/var/cache/apt/archives/perl-openssl-defaults_3_armhf.deb
/var/cache/apt/archives/po-debconf_1.0.21_all.deb
/var/cache/apt/archives/python3-debian_0.1.35_all.deb
/var/cache/apt/archives/python3-gpg_1.12.0-6_armhf.deb
/var/cache/apt/archives/python3-magic_2%3a0.4.15-2_all.deb
/var/cache/apt/archives/python3-unidiff_0.5.4-1_all.deb
/var/cache/apt/archives/squid-langpack_20190110-1_all.deb
/var/cache/apt/archives/t1utils_1.41-3_armhf.deb
/var/cache/apt/archives/wdiff_1.2.2-2_armhf.deb
pi@raspberrypi:~/websafety/core.debian10 $ 

```

**Comment by ra-at-diladele-com** [comment]
I will be able to take a look only after Jan 5 ☹

**Comment by ra-at-diladele-com** [comment]
Yes I confirm, I was not able to build Squid 4.9-2 from Debian 10 on RPI. Have no idea what how to pass the -latomic flag through the debian/rules file..

**Comment by BrandonSk** [comment]
Hi.

I have the same problem. I stumbled across [this post](https://root-forum.cern.ch/t/root-6-18-04-fails-to-compile-on-raspbian-10-with-solution/36514), but I have no idea how to actually do it. Since the guy there has the same problem during compiling on Raspberry PI (Raspbian), maybe it could work here too. Can you please check and give us feedback?

> The solution is to add the following cmake option:
> -DCMAKE_SHARED_LINKER_FLAGS=’-latomic’

Thanks.

**Comment by ra-at-diladele-com** [comment]
Thanks, I will take a look during 7.4 development - I actually tried this solution but somehow Debian rules did not applied the -latomic flag :( need to research more.

**Comment by BrandonSk** [comment]
Hello.

I think I did manage to apply the -latomic flag (at least the error no longer displays), but as said before, I still do not know exactly what I am doing :) but I try to analyze things and do trial-and-error approach.

So, I looked at the "rules" file located in directory: ~/websafety-master/core.debian10/build/squid/squid-4.10/debian/
There on the line 7 is this condition:
```
ifneq (,$(filter $(DEB_HOST_ARCH), armel m68k mips mipsel powerpc powerpcspe sh4))
        DEB_LDFLAGS_MAINT_APPEND += -latomic
endif
```
I suppose (I am not sure) that this condition is true, if my architecture is not in the list provided. Mine is not (I have armhf). But maybe it is the other way around... Nevertheless, I decided to apply the flag the hard way, so I changed the file to this:
```
ifneq (,$(filter $(DEB_HOST_ARCH), armel m68k mips mipsel powerpc powerpcspe sh4))
        DEB_LDFLAGS_MAINT_APPEND += -latomic
endif
DEB_LDFLAGS_MAINT_APPEND += -latomic
```

As I already ran the 03_squid.sh script before, I did not run it again, but instead I ran just this:
```
cd ~/websafety-master/core.debian10/build/squid/squid-4.10
dpkg-buildpackage -rfakeroot -b >> ~/build.output
```
I redirected the output so that I can attach here (but still I got some output in terminal and not everything was sent to the file). Long story short:

- The build process no longer complains about missng atomic link
- The process did finish with some complains about gpg I think and certificates generation(?) This cannot be seen in the file, but I can re-run the process and post here.
- In the output file there are lots of lines with double "//" in path. I do not know if this can cause some problems, just bringing it to the attention (just search for debian/tmp//usr).

So, if the above helps, I am happy. If you have a quick hint what I could try next, I will. But I will just as happily wait for version 7.4, as you wrote before (btw, any at least rough estimation on when this could be?).

Cheers,
Brandon.
[build.output.gz](https://github.com/diladele/websafety/files/4349419/build.output.gz)

**Comment by BrandonSk** [comment]
**IT COMPILED!**
...just an update to the previous post. In fact the compile process worked! The error at the end was just related to not being able to sign the package (missing private key).
The output was:
```
dpkg-buildpackage: info: binary-only upload (no source included)
 signfile squid_4.10-1_armhf.buildinfo
gpg: skipped "Luigi Gangitano <luigi@debian.org>": No secret key
gpg: dpkg-sign.fOFBpQ2q/squid_4.10-1_armhf.buildinfo: clear-sign failed: No secret key

dpkg-buildpackage: error: failed to sign .buildinfo file
```

### The working workaround for Raspbian
_Note: This is not a universal fix, but it does work for me on Raspbian and armhf architecture._

Before running the `03_squid.sh` script, modified it by adding 1 extra line as follows:
```
...
# modify configure options in debian/rules, add --enable-ssl --enable-ssl-crtd
patch squid-${SQUID_VER}/debian/rules < ../../rules.patch
sed -i '10i DEB_LDFLAGS_MAINT_APPEND += -latomic' squid-${SQUID_VER}/debian/rules

# build the package
cd squid-${SQUID_VER} && dpkg-buildpackage -rfakeroot -b

# and revert
popd
```
The new line is the one starting with "sed"

Then run the script as given in the guide. I already ran also script `04_install.sh` and squid got install.
I am going to continue with the process now. I hope it'll work at the end.

Hope this helps someone ;)

Cheers,
B.

PS: Then my logic says, that original problem is in the "ifneq" statement in the original 03_squid.sh script. But I am not sure how it should look like.

**Comment by ra-at-diladele-com** [comment]
Thanks a lot for your time! I will check and add the change into the 7.4 building scripts.

**Comment by gael1980** [comment]
Hello,
I'm confirming the patch is working.
Squid 4.10 works correctly on a 3b+ rpi.
but it is impossible to run the 05.sh file.

`Making all in src
make[1] : on entre dans le répertoire « /mnt/DD/websafety/core.debian10/ecap_clamav_adapter-2.0.0/src »
make  all-am
make[2] : on entre dans le répertoire « /mnt/DD/websafety/core.debian10/ecap_clamav_adapter-2.0.0/src »
/bin/bash ../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H    -I../src -I../src   -g -O1 -Wall -Wwrite-strings -Woverloaded-virtual -pipe -pthread -MT ecap_clamav_adapter
_la-Xaction.lo -MD -MP -MF .deps/ecap_clamav_adapter_la-Xaction.Tpo -c -o ecap_clamav_adapter_la-Xaction.lo `test -f 'Xaction.cc' || echo './'`Xaction.cc
libtool: compile:  g++ -DHAVE_CONFIG_H -I../src -I../src -g -O1 -Wall -Wwrite-strings -Woverloaded-virtual -pipe -pthread -MT ecap_clamav_adapter_la-Xaction.lo -MD -MP -MF .deps/ec
ap_clamav_adapter_la-Xaction.Tpo -c Xaction.cc  -fPIC -DPIC -o .libs/ecap_clamav_adapter_la-Xaction.o
In file included from Xaction.cc:11:
Service.h:99:45: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
     void finalizeTricklingConfig(const std::auto_ptr<TricklingConfig> &oldConfig);
                                             ^~~~~~~~
In file included from /usr/include/c++/8/memory:80,
                 from /usr/include/c++/8/tr1/memory:39,
                 from /usr/include/libecap/common/memory.h:7,
                 from /usr/include/libecap/common/area.h:7,
                 from FileBuffer.h:8,
                 from Xaction.cc:7:
/usr/include/c++/8/bits/unique_ptr.h:53:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
Xaction.cc: In member function ‘virtual libecap::Area Adapter::Xaction::abContent(libecap::size_type, libecap::size_type)’:
Xaction.cc:305:49: error: no matching function for call to ‘min(libecap::size_type&, const Size&)’
         bufSize = std::min(bufSize, tricklingMax);
                                                 ^
In file included from /usr/include/c++/8/list:60,
                 from Answers.h:9,
                 from Xaction.cc:6:
/usr/include/c++/8/bits/stl_algobase.h:195:5: note: candidate: ‘template<class _Tp> constexpr const _Tp& std::min(const _Tp&, const _Tp&)’
     min(const _Tp& __a, const _Tp& __b)
     ^~~
/usr/include/c++/8/bits/stl_algobase.h:195:5: note:   template argument deduction/substitution failed:
Xaction.cc:305:49: note:   deduced conflicting types for parameter ‘const _Tp’ (‘unsigned int’ and ‘Size’ {aka ‘long long unsigned int’})
         bufSize = std::min(bufSize, tricklingMax);
                                                 ^
In file included from /usr/include/c++/8/list:60,
                 from Answers.h:9,
                 from Xaction.cc:6:
/usr/include/c++/8/bits/stl_algobase.h:243:5: note: candidate: ‘template<class _Tp, class _Compare> constexpr const _Tp& std::min(const _Tp&, const _Tp&, _Compare)’
     min(const _Tp& __a, const _Tp& __b, _Compare __comp)
     ^~~
/usr/include/c++/8/bits/stl_algobase.h:243:5: note:   template argument deduction/substitution failed:
Xaction.cc:305:49: note:   deduced conflicting types for parameter ‘const _Tp’ (‘unsigned int’ and ‘Size’ {aka ‘long long unsigned int’})
         bufSize = std::min(bufSize, tricklingMax);
                                                 ^
In file included from /usr/include/c++/8/algorithm:62,
                 from Xaction.cc:19:
/usr/include/c++/8/bits/stl_algo.h:3450:5: note: candidate: ‘template<class _Tp> constexpr _Tp std::min(std::initializer_list<_Tp>)’
     min(initializer_list<_Tp> __l)
     ^~~
/usr/include/c++/8/bits/stl_algo.h:3450:5: note:   template argument deduction/substitution failed:
Xaction.cc:305:49: note:   mismatched types ‘std::initializer_list<_Tp>’ and ‘unsigned int’
         bufSize = std::min(bufSize, tricklingMax);
                                                 ^
In file included from /usr/include/c++/8/algorithm:62,
                 from Xaction.cc:19:
/usr/include/c++/8/bits/stl_algo.h:3456:5: note: candidate: ‘template<class _Tp, class _Compare> constexpr _Tp std::min(std::initializer_list<_Tp>, _Compare)’
     min(initializer_list<_Tp> __l, _Compare __comp)
     ^~~
/usr/include/c++/8/bits/stl_algo.h:3456:5: note:   template argument deduction/substitution failed:
Xaction.cc:305:49: note:   mismatched types ‘std::initializer_list<_Tp>’ and ‘unsigned int’
         bufSize = std::min(bufSize, tricklingMax);
                                                 ^
make[2]: *** [Makefile:561: ecap_clamav_adapter_la-Xaction.lo] Error 1
make[2] : on quitte le répertoire « /mnt/DD/websafety/core.debian10/ecap_clamav_adapter-2.0.0/src »
make[1]: *** [Makefile:391: all] Error 2
make[1] : on quitte le répertoire « /mnt/DD/websafety/core.debian10/ecap_clamav_adapter-2.0.0/src »
make: *** [Makefile:406: all-recursive] Error 1
root@DietPi:/mnt/DD/websafety/core.debian10/ecap_clamav_adapter-2.0.0#`

**Comment by Lnxzz** [comment]
In order to get the ClamAV adapter to compile, you should apply the following patch:

wget -q https://bugs.launchpad.net/ecap/+bug/1752929/+attachment/5068296/+files/min-deduction-v2p0p0.patch
# apply patch for bugfix
cd ecap_clamav_adapter-2.0.0
patch -p1 < ../min-deduction-v2p0p0.patch

**Comment by ra-at-diladele-com** [comment]
Updated the https://github.com/diladele/websafety/blob/release-7.4.0/core.debian10/rules.patch - seems to compiled normally and Squid 4.10 is running.

**Comment by ra-at-diladele-com** [comment]
Added https://github.com/diladele/websafety/blob/release-7.4.0/core.debian10/Xaction.cc.patch - seems to compile normally now. Closing.
